drivers/sensor/apds9253/apds9253.c

```diff
@@ -121,9 +121,98 @@ static int apds9253_channel_get(const struct device *dev, enum sensor_channel ch
 	return 0;
 }
 
+static int apds9253_attr_set_gain(const struct device *dev, uint8_t gain)
+{
+	const struct apds9253_config *config = dev->config;
+	struct apds9253_data *drv_data = dev->data;
+	uint8_t value;
+
+	if (drv_data->gain == gain) {
+		return 0;
+	}
+
+	static uint8_t value_map[] = {
+		APDS9253_LS_GAIN_RANGE_1, APDS9253_LS_GAIN_RANGE_3,  APDS9253_LS_GAIN_RANGE_6,
+		APDS9253_LS_GAIN_RANGE_9, APDS9253_LS_GAIN_RANGE_18,
+	};
+
+	if (gain < APDS9253_LS_GAIN_RANGE_1 || gain > APDS9253_LS_GAIN_RANGE_18) {
+		return -EINVAL;
+	}
+	value = value_map[gain];
+
+	if (i2c_reg_update_byte_dt(&config->i2c, APDS9253_LS_GAIN_REG, APDS9253_LS_GAIN_MASK,
+				   (value & APDS9253_LS_GAIN_MASK))) {
+		LOG_ERR("Not able to set light, sensor gain is not set");
+		return -EIO;
+	}
+
+	drv_data->gain = gain;
+
+	return 0;
+}
+
+static int apds9253_attr_set_sampl_freq(const struct device *dev,
+					const struct sensor_value *sampl_freq)
+{
+	const struct apds9253_config *config = dev->config;
+	struct apds9253_data *drv_data = dev->data;
+	uint8_t period_val;
+	float freq_hz;
+
+	freq_hz = sampl_freq->val1 + (sampl_freq->val2 / 1000000);
+
+	if (freq_hz == 0.5f) {
```

Comment:
This floating point compare will never actually be `true`, please don't use a `float` type for `freq_hz`. As it stands even if you had `val1 == 0` and `val2 == 999999` you'd get `freq_hz` as 0 with the calculation above. I suggest using mHz as your units

```c
uint32 mhz = saml_freq->val1 * 1000 + (sampl_freq->val2 / 1000);

if (mhz >= 40000) {
  period_val = APDS9253_LS_MEAS_RATE_MES_25MS;
} else if (mhz >= 20000) {
  period_val = APDS9253_LS_MEAS_RATE_MES_50MS;
} ... {
} else if (mhz >= 500) {
  preiod_val = APDS9253_LS_MEAS_RATE_MES_2000MS;
} else {
  // Turn off the sensor?
}
```